### DISCUSSION TITLE
I have a derived image that works with 0.9 but it crashes from 0.10 to the latest due to "exec format error"

### DISCUSSION THREAD

**Discussion by ignlg** [burningalchemist/sql_exporter#486] [Q&A] [answered]
I have this Dockerfile that uses `sql_exporter` as the base:

```
FROM burningalchemist/sql_exporter:0.14
USER 0
COPY ./docker-entrypoint.sh /docker-entrypoint.sh
CMD chmod +x /docker-entrypoint.sh

USER nobody

ENTRYPOINT [ "/docker-entrypoint.sh" ]
CMD [ "/bin/sql_exporter" ]
```

This has been running smoothly for a year with v0.9.

I was testing to update it to 0.14, but it crashed. I tracked it down and I can only go up to v0.9 without the issue. If I update it with version 0.10 or higher, the container does not run, with the error:

```
exec /docker-entrypoint.sh: exec format error
```

It is not about the content of the script. I checked it and it is not executed at all.

What could be happening?

**Comment by burningalchemist** [comment, marked as answer]
Hi @ignlg, what is the platform you're running your script on? Does the native `sql_exporter` image work for you?

Thinking a bit more, since `v0.10` sql_exporter is using `Dockerfile.multi-arch` file for building images (to support linux/amd64 and linux/arm64 [platforms](https://hub.docker.com/r/burningalchemist/sql_exporter/tags)), so it might be that you need to keep in mind the platform-specific parameters and build your image accordingly with passing these.

TL;DR: sql_exporter binary from the original image might be unsupported by the destination image platform/os. The best way to find this out is to extract the binary, and check info with `file sql_exporter` to get the binary headers.

References:
- https://github.com/burningalchemist/sql_exporter/blob/master/Dockerfile.multi-arch
- https://docs.docker.com/build/building/multi-platform/

**Reply by ignlg** [reply]
I will check if the image builder and runner are different architectures. I was wrongly reviewing the history of the `Dockerfile` looking for the answer. Thank you for your great reply, @burningalchemist.

**Reply by burningalchemist** [reply]
No worries, I'm happy to help, let me know if you have resolved the issue (this helps with building the knowledge base for other users). 👍

I'd say, based on your final image, it might make sense to copy `sql_exporter` binary somewhere else instead (since it's just a single binary). Just a random idea. :)

**Reply by ignlg** [reply]
Finally, I am copying the binary file into an Alpine image and setting everything there. No more build issues, and 0.14 is up and running :)

**Reply by burningalchemist** [reply]
Perfect, thanks for reporting the status back! Hope it works fine for you! 🚀
